### DISCUSSION TITLE
is there a bug for required attribute when create?

### DISCUSSION THREAD

**Discussion by FauziFadhi** [sequelize/sequelize#14219] [Help & Questions] [answered]
"sequelize": "^6.17.0"

i think there is a bug for creationAttributes, all of my defined interface attributes become optional

I try to prove some interface, and there is a difference between the docs of code and the actual one
![image](https://user-images.githubusercontent.com/22959762/157592939-3b94aa74-e4f3-4773-8c1b-8a8081bc3be7.png)

and all of my defined interface attributes becomes optional 
![image](https://user-images.githubusercontent.com/22959762/157593322-d4f23036-eb09-4f97-8e1d-f3e7ef563741.png)

or did I miss something?

**Comment by ephys** [comment, marked as answer]
We have tests that checks for that https://github.com/sequelize/sequelize/blob/main/test/types/infer-attributes.ts#L52 so it shouldn't be the case

Note that if [`exactOptionalPropertyTypes`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html#exact-optional-property-types---exactoptionalpropertytypes) is not enabled, then `lastName` will be considered nullish too

I'd be interested to look at your model definition to see which field is incorrectly optional :)

**Reply by FauziFadhi** [reply]
Ohh. Thank you very much.

I am just checking the code here and here. but not checking the tsconfig. you really help me a lot.
